initial draft of mosfet in semiconductor.json for spec v0.7

rsnancollas opened the pull request:
A little late, but here is an initial draft of the mosfet spec based on our discussion from a few meetings ago.

rsnancollas committed:
initial draft of mosfet in semiconductor.json for spec v0.7

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,106 @@
+                "partType",
+                "manufacturer",
+                "mpn",
+                "datasheetVersion",
```
required should be componentID + the other 2 properties

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,106 @@
+                    "$ref": "definitions.json#/unit"
+                },
+                "vgsThMax": {
+                    "description": "maximum gate to source voltage difference required to produce a conducting path between drain and source",
```
Should we change this to vgsThresholdMax?

rsnancollas committed:
made changes to part-spec-proto to align with current (0.5) spec (#22)

* made changes to part-spec-proto to align with current (0.5) spec

* added package property to part-spec-proto files

* addressed review feedback on part-spec-proto

rsnancollas committed:
added diode to semiconductor change

asadate committed:
Update semiconductor.json

Updated mosfet properties

asadate committed:
Update semiconductor.json

Added more properties for the diode

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,318 @@
+                },
+                "ifm": {
+                    "description": "maximum continuous forward current a diode can support",
+                    "comment": "units of amps",
```
"forward current of the device"  instead of " forward current of a diode."

asadate replied:
Changes will be done in a future revision. Updating now will require us to go update all the other component datasheets already released.

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,318 @@
+                    "$ref": "graph.json#/graphDefiniton"
+                },
+                "esdProtection": {
+                    "description": "whether a diode has built in esd protection",
```
esdProtection property not needed

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,318 @@
+                    "items": {
+                        "$ref": "definitions.json#/pinSpec"
+                    }
+                }
```
Missing properties: Number of diodes, Number of protected lines, Configuration

asadate replied:
Added

asadate commented on part-spec/semiconductor.json:
```diff
@@ -0,0 +1,318 @@
+                    "enum": [
+                        "nType",
+                        "pType"
+                    ],
```
add nAndpType in the enum list

asadate replied:
changed

asadate committed:
Update semiconductor.json

Updates to capture Nexperia input

asadate reviewed (commented):
Changes requested by Nexperia were added.

asadate committed:
Updated file to add "limiting value".

Some FETs properties are limiting values. Description of these properties were updated to reflect this.

asadate commented on part-spec/definitions.json:
```diff
@@ -496,5 +496,27 @@
     }
+  },
+  "conditionalProperty": {
+    "type": "object",
```
Will remove "min", "max", "typ", from here and replace with "value" following up WG discussion to call out minValue,..maxValue as different properties.

asadate committed:
Updated conditionalProperty object

Removed the "min", "max", "typ" properties and replaced with "value". Each of those properties should be called through separate conditonalProperty objects.

asadate committed:
Updated diode to add Min, Typ, Max properties

asadate committed:
Updated FET with min, typ, max properties

Per work group discussions, it's better for tools if min, typ, max properties are called out separately in the e-datasheets.

asadate committed:
Added examples for: diode, FET and, retimer

asadate committed:
corrected typo

asadate committed:
Added USB type-c PD Controller example

asadate committed:
Added an example for the microcontroller/EC 

Fixed previous issues on the complex pin examples.

asadate commented on support-docs/examples.md:
```diff
@@ -431,3 +431,809 @@ Buck Example
+            "unitFactor": 1e-3,
+            "valueDefined": true
+        },
+        "conditions": ["forward current (If)=800mA"]     
```
Add JSON format easy to parse

Carrotman42 reviewed (approved):
I took a look and everything seems OK. To me, that implies that maybe I don't have the ability to give a good review here ;)

But you can at least count this as a looks-good-to-kevin.

asadate committed:
Changes on conditional property 

Changes on how the conditional property is written: to make it more friendly for tools.

asadate committed:
Fixed typos & added properties

fixed typos and added definitions of some of the properties referred to in graph definition.

asadate reviewed (approved):
Requested changes implemented
